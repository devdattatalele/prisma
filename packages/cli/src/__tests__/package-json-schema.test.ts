import { jestConsoleContext, jestContext } from '@prisma/get-platform'
import { getSchemaWithPath } from '@prisma/internals'

const ctx = jestContext.new().add(jestConsoleContext()).assemble()

it('load schema from package.json and generate a deprecation warning', async () => {
  ctx.fixture('package-json-schema')

  await expect(getSchemaWithPath()).resolves.toMatchObject({
    schemaRootDir: expect.stringMatching(/custom-path$/),
  })

  expect(ctx.mocked['console.warn'].mock.calls.join('\n')).toMatchSnapshot()
})
