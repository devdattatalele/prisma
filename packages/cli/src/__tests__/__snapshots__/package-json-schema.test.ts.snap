// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load schema from package.json and generate a deprecation warning 1`] = `
"prisma:warn The "prisma.schema" property in your package.json is deprecated and will be removed in a future release.
Please use prisma.config.ts instead.
For more information on this change, visit: https://www.prisma.io/docs/orm/reference/prisma-config-reference"
`;
